ospf6d/ospf6_message.c

```diff
@@ -2356,7 +2356,20 @@ static void ospf6_send_lsupdate(struct ospf6_neighbor *on,
 	}
 	if (oi) {
 		ospf6_packet_add(oi, op);
-		OSPF6_MESSAGE_WRITE_ON(oi);
+		/* If ospf instance is being deleted, send the packet
+		 * immediately */
+		if (oi->area && oi->area->ospf6
+		    && oi->area->ospf6->inst_shutdown) {
```

Comment:
ospf6_flood_interface function already takes care of sending the LSAs immediately if instance is getting deleted and is having the same check. Please re-check once.

Reply:
ospf6_lsupdate_send_neighbor_now() is called from ospf6_flood_interface() which checks for inst_shutdown and the write thread is added in ospf6_send_lsupdate().  Since the ospf instance is getting deleted, ospf6_write() should be called directly instead of adding write thread. The issue observed is that LSAs does not get deleted immediately from peer router database. it gets deleted after timeout.  Therefore calling write()  directly is required.  Similar functionality is present in ospfv2 code ospf_ls_upd_queue_send() function

Reply:
Okay got it. The changes looks fine to me.